data_structures/src/superblock.rs

```diff
@@ -327,26 +312,43 @@ pub fn mining_build_superblock(
     ars_ordered_hash_leaves: &[Hash],
     index: u32,
     last_block_in_previous_superblock: Hash,
-) -> Option<SuperBlock> {
-    let last_block = block_headers.last()?.hash();
-    let merkle_drs: Vec<Hash> = block_headers
-        .iter()
-        .map(|b| b.merkle_roots.dr_hash_merkle_root)
-        .collect();
-    let merkle_tallies: Vec<Hash> = block_headers
-        .iter()
-        .map(|b| b.merkle_roots.tally_hash_merkle_root)
-        .collect();
-
-    Some(SuperBlock {
-        ars_length: ars_ordered_hash_leaves.len() as u64,
-        data_request_root: hash_merkle_tree_root(&merkle_drs),
-        tally_root: hash_merkle_tree_root(&merkle_tallies),
-        ars_root: hash_merkle_tree_root(ars_ordered_hash_leaves),
-        index,
-        last_block,
-        last_block_in_previous_superblock,
-    })
+) -> SuperBlock {
+    let last_block = block_headers.last();
+    match last_block {
+        None =>
+        // Build "empty" Superblock (there were no blocks during super-epoch)
+        {
+            SuperBlock {
+                ars_length: ars_ordered_hash_leaves.len() as u64,
+                ars_root: hash_merkle_tree_root(ars_ordered_hash_leaves),
+                index,
+                last_block: last_block_in_previous_superblock,
+                last_block_in_previous_superblock,
+                ..Default::default()
```

Comment:
Using `Default::default()` here sets the remaining fields to `Hash::SHA256([0; 32])`. Currently the hash of an empty merkle tree is set to the empty hash `e3b0c44298f...`. It sure makes sense to use `00000000...` as the default hash, but we should be consistent.

Reply:
Agreed! It was seen but not changed...